### PR TITLE
Add beatmaps section to user profiles

### PR DESCRIPTION
**Pull request by nekodex**
Adds your favourite beatmaps and your ranked/approved beatmaps to your user profile. (doesn't paginate yet, so may be comical)

The beatmap preview images currently look like buttes, but that'll be fixed when we get around to generating proper thumbnails.

### CODE REVIEW

File: resources/assets/coffee/react/beatmaps/panel.coffee [hunk 1]
```diff
@@ -24,33 +24,33 @@ class @Panel extends React.Component
...
+      if beatmap.difficulties.data.length > 5
+        difficulties.push el(BeatmapDifficultyIcon, difficulty: beatmap.difficulties.data[0], key: 0)
+        difficulties.push span key: 'over', '+' + (beatmap.difficulties.data.length - 2)
       else
-        for difficulty, index in beatmap.difficulties
+        for difficulty, index in beatmap.difficulties.data
           difficulties.push el(BeatmapDifficultyIcon, difficulty: difficulty, key: index)
 
-    div href: '/beatmaps/modding/'+beatmap.beatmapset_id, className: 'beatmap object_link shadow-hover', objectid: beatmap.beatmapset_id,
+    div href: '#', className: 'beatmap object_link shadow-hover', objectid: beatmap.beatmapset_id,
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmaps/panel.coffee]: this a what

File: resources/assets/coffee/react/profile-page/beatmaps.coffee [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+###
+
+{div, h2, h3, ul, li, a, p, pre, span} = React.DOM
+el = React.createElement
+
+ProfilePage.Beatmaps = React.createClass
+  mixins: [React.addons.PureRenderMixin]
+
+  render: ->
+    beatmaps = {
```
Comment (nanaya) [inline resources/assets/coffee/react/profile-page/beatmaps.coffee]: so javascripty :D 

```coffeescript
beatmaps =
  favourite: herp
  rankedetc: derp
```

File: resources/assets/coffee/react/beatmaps/panel.coffee (hunk 1 above)
Comment (nekodex) [inline resources/assets/coffee/react/beatmaps/panel.coffee]: ( ͡° ͜ʖ ͡°)

File: resources/assets/coffee/react/profile-page/beatmaps.coffee (hunk 2 above)
Comment (nekodex) [inline resources/assets/coffee/react/profile-page/beatmaps.coffee]: not enough coffee today

File: resources/assets/coffee/react/profile-page/beatmaps.coffee [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      'favourite': @props.favouriteBeatmaps?.data || [],
+      'ranked_and_approved': @props.rankedAndApprovedBeatmaps?.data || []
+    }
+
+    div
+      className: 'osu-layout__row osu-layout__row--page profile-extra'
+      div className: 'profile-extra__anchor js-profile-page-extra--scrollspy', id: 'beatmaps'
+      h2 className: 'profile-extra__title', Lang.get('users.show.extra.beatmaps.title')
+      ['favourite', 'ranked_and_approved'].map (section) ->
```
Comment (nanaya) [inline resources/assets/coffee/react/profile-page/beatmaps.coffee]: there probably is something like ruby's `beatmaps.map { |key, value| ... }`

File: resources/assets/coffee/react/profile-page/beatmaps.coffee [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        div null,
+          h3 className: 'profile-extra__title--small', Lang.get('users.show.extra.beatmaps.' + section + '.title', count: beatmaps[section].length)
```
Comment (nanaya) [inline resources/assets/coffee/react/profile-page/beatmaps.coffee]: `"derp.#{herp}.welp"`

File: resources/assets/coffee/react/profile-page/beatmaps.coffee (hunk 3 above)
Comment (nekodex) [inline resources/assets/coffee/react/profile-page/beatmaps.coffee]:
```
¯\_(ツ)_/¯
```

**Pull request merged**